### ISSUE TITLE
False positive MissingTemplateParam

### ISSUE THREAD

**Issue by va5ja** [vimeo/psalm#9208]
https://psalm.dev/r/8cec79421f

```
<?php

class C implements \IteratorAggregate {
    public function getIterator(): \ArrayIterator {
        return new \ArrayIterator([]);
    }
}
```

The above code was fine in older Psalm versions (^4). Do I always need to add something like below?
```
/**
 * @template T
 * @implements \IteratorAggregate<T>
 */
 ```

**Comment by weirdan** [comment]
> Do I always need to add something like below?
> ```php
> /**
>  * @template T
>  * @implements \IteratorAggregate<T>
>  */
> ```

Yes.
